### PR TITLE
ID-2782 [Feat] Cache data if date range hasn't changed

### PR DESCRIPTION
**Pull request by tonytlwu**
Ref. https://weboo.atlassian.net/browse/ID-2782

Requires https://github.com/Fliplet/fliplet-api/pull/5594

### CODE REVIEW

**Review by squallstar — APPROVED** [review]

File: src/services/analytics.js
Comment (squallstar) [inline src/services/analytics.js:16]: How much data are we talking about? Just checking it's not too large to be cached.

Comment (tonytlwu) [reply src/services/analytics.js:16]: @squallstar This is the request that we will be caching. It seems that there's possibly a backend optimisation that can be made too.

I'll wait for your confirmation before merging the PRs for this work.

image

image

Comment (squallstar) [reply src/services/analytics.js:16]: @tonytlwu thanks, I can see that the payload is relatively small so I'm good with that.

In regards to optimising this in the backend, can you send me the request body and full endpoint (including appId and environment) for that request so that I can look into it?

Comment (tonytlwu) [reply src/services/analytics.js:16]: Added to Jira as comment

**Pull request merged**
